src/main/java/mong/Mong.java

```diff
@@ -0,0 +1,358 @@
+import mong.task.Deadline;
+import mong.task.Event;
+import mong.task.Task;
+import mong.task.TaskType;
+import mong.task.Todo;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+public class Mong {
+    public static final String FILE_PATH = "src/main/java/mong/data/mong.txt";
+    public static ArrayList<Task> list = new ArrayList<>();
+    public static final int LENGTH_DEADLINE = 8;
+    public static final int LENGTH_BY = 3;
+    public static final int LENGTH_TODO = 4;
+    public static final int LENGTH_EVENT = 5;
+    public static final int LENGTH_FROM = 4;
+    public static final int LENGTH_TO = 3;
+    public static final String HORIZONTAL_LINE = "--------------------------------------------------";
+
+    /**
+     * Writes new content to txt file.
+     */
+    private static void writeToFile(String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(FILE_PATH);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    /**
+     * Formats latest list to txt file format.
+     */
+    public static String parseListToTxt() {
+        StringBuilder textToAdd = new StringBuilder();
+        for (Task task : list) {
+            textToAdd.append(task.toFileFormat()).append(System.lineSeparator());
+        }
+        return textToAdd.toString();
+    }
+
+    /**
+     * Adds tasks line by line from txt file.
+     */
+    public static void loadLineContents(String line) {
+        String[] lineData = line.split(" \\| ");
+        if (lineData[0].contentEquals("T")) {
+            addTodo(lineData[1], lineData[2]);
+        } else if (lineData[0].contentEquals("D")) {
+            addDeadline(lineData[1], lineData[2], lineData[3]);
+        } else if (lineData[0].contentEquals("E")) {
+            addEvent(lineData[1], lineData[2], lineData[3], lineData[4]);
+        }
+    }
+
+    /**
+     * Adds all tasks in txt file.
+     */
+    public static void loadFileContents(String filePath) throws FileNotFoundException {
+        File file = new File(filePath);
+        Scanner scanner = new Scanner(file);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            loadLineContents(line);
+        }
+    }
+
+    /**
+     * Marks item in index as completed.
+     */
+
+    public static void mark(String input) {
+        // the itemIndex is -1 than the input from the user
+        int itemIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+        if (itemIndex < 0 || itemIndex >= list.size()) {
+            throw new NullPointerException();
+        }
+        try {
+            list.get(itemIndex).setCompleted(true);
+            System.out.println("Mong >_<!!I have marked it as completed:");
+            System.out.println(list.get(itemIndex));
+        } catch (NullPointerException | IndexOutOfBoundsException e) {
```

Comment:
Currently, the mark and unmark methods throw a NullPointerException or IndexOutOfBoundsException when the index is invalid. It would be better to handle these cases before calling the setCompleted method to improve clarity.